backend/routes/graph.py

```
@@ -8,16 +10,25 @@
 router = APIRouter()
+logger = logging.getLogger(__name__)
 
 
 @router.get("/{user_id}")
 def get_user_graph(user_id: str):
-    return get_graph(user_id)
+    try:
+        return get_graph(user_id)
+    except Exception as e:
+        logger.exception("get_graph failed for user_id=%s", user_id)
+        raise HTTPException(status_code=500, detail=str(e) or "Failed to fetch graph")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)